src/locking.c

```diff
@@ -6,35 +6,40 @@
 #include "common.h"
 #include "types.h"
 #include "locking.h"
+#include "shared.h"
 
 #if defined (F_SETLKW)
 
-int lock_file (FILE *fp)
+int hc_lockfile (HCFILE *fp)
 {
+  if (!fp) return -1;
+
   struct flock lock;
 
   memset (&lock, 0, sizeof (struct flock));
 
   lock.l_type = F_WRLCK;
 
   /* Needs this loop because a signal may interrupt a wait for lock */
-  while (fcntl (fileno (fp), F_SETLKW, &lock))
+  while (fcntl (hc_fileno (fp), F_SETLKW, &lock))
   {
     if (errno != EINTR) return -1;
   }
 
   return 0;
 }
 
-int unlock_file (FILE *fp)
+int hc_unlockfile (HCFILE *fp)
 {
+  if (!fp) return -1;
```

Comment:
Same here and probably other places, too. The idea is to make it easier to read by not need to interpret NOT while reading.

Reply:
y